### PR TITLE
Add header and footer section groups

### PR DESCRIPTION
**Pull request by utkarshsaxena93**
### PR Summary: 
Introduce section groups by adding a Header section group and a footer section group.

You can now create section groups in your theme’s layout. Section groups are JSON containers that allow merchants to add, remove, and reorder sections in areas of the layout file such as the header and footer.

To include a section group in a layout file, add the section group to the **sections** directory of your theme, then add a `{% sections '<filename>' %}` Liquid tag that references your new sections file to the layout file where you want it to be rendered.

### Why are these changes introduced?

New feature.

### Visual impact on existing themes
There should be no visual impact on themes. However in the Editor you can now add new sections within these section groups allowing you to add more customization. 

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139224678422/editor)

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [x] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: sections/footer-group.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          "settings": {
+            "heading": "Info",
+            "menu": "footer"
+          }
+        },
+        "footer-2": {
+          "type": "text",
+          "settings": {
+            "heading": "Our mission",
+            "subtext": "<p>Share contact information, store details, and brand content with your customers.<\/p>"
```
Comment (utkarshsaxena93) [inline sections/footer-group.json:26]: Should these not refer to translations instead of hardcoding a string in English?

Comment (LucasLacerdaUX) [reply sections/footer-group.json:26]: I don't think we support translations as settings values, so our default content is always using English.

**Review by eugenekasimov — APPROVED** [review]

**Comment by mmorissette** [comment]
I wonder if we should wait for https://github.com/Shopify/shopify/issues/397729 to be addressed before merging this PR so not to break the current development workflow for flagship theme developers. WDYT @utkarshsaxena93 and @veken1199

**Review by melissaperreault — APPROVED** [review]
Thank you! Assuming testing the zip should be the same, I tested and everything works as expected! 🚀

**Comment by bakura10** [comment]
Michael from Maestrooo here.

For our themes we decided to disable the sections on the header group but keep them on footer group. While it makes little sense to add a big section in the header, in the footer this makes sense (think of a text with icon section to show trust values, or a image with text overlay before the footer to have it on every pages…)

however I would prefer to align with Dawn behaviour to makes it more consistent. What is the motivation behind blocking all sections in footer group?

**Comment by blanklob** [comment]
Haha I was about to make a PR then found out you did It.

**Comment by melissaperreault** [comment]
Hey @bakura10 ! I can help answer the following!

> What is the motivation behind blocking all sections in footer group?

We only limited these until we learn more how merchants are using groups and sections. It is easier to add than to remove! I agree that most sections are good addition for the footer. Hope this helps you!

**Pull request merged**